spring-vault-core/src/main/java/org/springframework/vault/core/VaultKeyValue2Template.java

```diff
@@ -89,4 +90,29 @@ public void put(String path, Object body) {
 		doWrite(createDataPath(path), Collections.singletonMap("data", body));
 	}
 
+	/**
+	 * Performs a KV Patch operation.
+	 * @param path must not be {@literal null} or empty.
+	 * @param kv New key value map to be updated
+	 * @since 2.3
+	 */
+	public void patch(String path, Map<String, ?> kv) {
+		Assert.hasText(path, "Path must not be empty");
+
+		// To do patch operation, we need to do a read operation first
+		VaultResponse readResponse = get(path);
+		Assert.notNull(readResponse, "VaultResponse must not be null");
```

Comment:
If the secret doesn't exist, a meaningful exception should be thrown. I think a new `SecretNotFoundException` type (`extends VaultException`) would make sense.

Reply:
Applied here: https://github.com/spring-projects/spring-vault/pull/587/commits/95d25bd9120c340cecc2eab71c67ffc047e41f07#diff-917f198074796659086e92ad34820ddfR106-R112